src/Symfony/Component/Validator/ConstraintValidator.php

```diff
@@ -87,8 +87,14 @@ protected function formatValue($value, $format = 0)
     {
         if (($format & self::PRETTY_DATE) && $value instanceof \DateTimeInterface) {
             if (class_exists('IntlDateFormatter')) {
+                $timezone = $value->getTimezone();
```

Comment:
Thank you for working on this. I looked about it and it can actually be resolved just by using the `setTimezone()` setter instead of passing the timezone in the constructor (cf https://3v4l.org/kDLqp). I think this is a better solution. WDYT?

Reply:
It’s definitely a much cleaner solution. I still believe that Z needs to be handled as a timezone for validation but maybe this is the applications problem and not the validator.

Reply:
Handling it is definitely better. But I'm not sure it's our responsibility to transform incompatible timezones between datetimes and the Intl date formatter. Let's wait for more opinions.

If we do it we just need to convert `Z` to `UTC` then (instead of falling back on the default timezone), isn'it?